src/commands/edge_app_server.rs

```diff
@@ -96,39 +102,75 @@ impl Default for Metadata {
 
 #[derive(Debug, Default, Deserialize)]
 struct MockData {
+    metadata: Metadata,
+    settings: HashMap<String, Value>,
+}
+
+#[derive(Debug, Default, Deserialize, Clone)]
+struct MockDataStr {
     metadata: Metadata,
     settings: HashMap<String, String>,
 }
 
+impl MockData {
+    fn new_from_str(mock_data_str: &MockDataStr) -> Self {
```

Comment:
this function also needs to be tested especially considering that you parse it manually. You need to test it with invalid input/edge cases.